### PR TITLE
[Test] Create new per-test fixtures

### PR DESCRIPTION
**Pull request by Warrows**
This makes a new basic testing fixture initializing less stuff for most test cases. It also reinitializes the fixture between each test suite.
Backport of bitcoin#5852 and bitcoin#5883.

### CODE REVIEW

**Comment by Warrows** [comment]
I was unable to reproduce the problem but I think these new changes should fix it.

**Comment by Warrows** [comment]
Rebased.

File: src/main.cpp [hunk 1]
```diff
@@ -5302,10 +5302,31 @@ bool CVerifyDB::VerifyDB(CCoinsView* coinsview, int nCheckLevel, int nCheckDepth
...
+    nBlockSequenceId = 1;
+    mapBlockSource.clear();
+    mapBlocksInFlight.clear();
+    nQueuedValidatedHeaders = 0;
+    nPreferredDownload = 0;
+    setDirtyBlockIndex.clear();
+    setDirtyFileInfo.clear();
+    mapNodeState.clear();
+
+    BOOST_FOREACH(BlockMap::value_type& entry, mapBlockIndex) {
```
Comment (Fuzzbawls) [inline src/main.cpp]: replace with a c++11 `for` loop

Comment (Warrows) [reply src/main.cpp]: Nice catch.

**Comment by Mrs-X** [comment]
`test/functional/test_runner.py --extended` shows still failed tests, I'll let it run later today without the `--extended`:

`> test/functional/test_runner.py --extended`:
Tests-01

**Comment by Fuzzbawls** [comment]
@Mrs-X this PR has little to do with the python functional regression tests, but everything to do with the Unit tests. We already know and accept that there are a number of functional tests that aren't passing currently (mainly because they haven't been ported to our specific environment yet).

**Comment by Mrs-X** [comment]
`test/functional/test_runner.py` doesn't look much better:
Tests-01

How did you call test_runner.py to have all tests succeed?

**Comment by Mrs-X** [comment]
Oh, you're right, I forgot that, it's been a while since my first check for this PR :man_facepalming: 
All good, merging...

**Pull request merged**